### ISSUE TITLE
Error upgrading from 1.10.0 to 1.11.0 - ArgumentError: wrong number of arguments (given 0, expected 1+) 

### ISSUE THREAD

**Issue by raivil** [gregnavis/active_record_doctor#132]
Bumping from 1.10.0 to 1.11.0 resulted in the error below.
I'm not sure if this is a gem issue or if it could be with my app.
Still investigating.

Any clues or suggestions to track down this issue?

```
ArgumentError: wrong number of arguments (given 0, expected 1+)
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/extraneous_indexes.rb:91:in `count'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/extraneous_indexes.rb:91:in `prefix?'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/extraneous_indexes.rb:82:in `replaceable_with?'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/extraneous_indexes.rb:40:in `block (4 levels) in subindexes_of_multi_column_indexes'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/extraneous_indexes.rb:39:in `select'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/extraneous_indexes.rb:39:in `block (3 levels) in subindexes_of_multi_column_indexes'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:213:in `block (3 levels) in each_index'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/logger/dummy.rb:7:in `log'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:96:in `log'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:212:in `block (2 levels) in each_index'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:205:in `each'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:205:in `block in each_index'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/logger/dummy.rb:7:in `log'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:96:in `log'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:204:in `each_index'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/extraneous_indexes.rb:38:in `block (2 levels) in subindexes_of_multi_column_indexes'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:277:in `block (3 levels) in each_table'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/logger/dummy.rb:7:in `log'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:96:in `log'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:276:in `block (2 levels) in each_table'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:271:in `each'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:271:in `block in each_table'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/logger/dummy.rb:7:in `log'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:96:in `log'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:270:in `each_table'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/extraneous_indexes.rb:37:in `block in subindexes_of_multi_column_indexes'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/logger/dummy.rb:7:in `log'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:96:in `log'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/extraneous_indexes.rb:36:in `subindexes_of_multi_column_indexes'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/extraneous_indexes.rb:31:in `detect'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:53:in `block in run'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/logger/dummy.rb:7:in `log'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:96:in `log'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:49:in `run'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/detectors/base.rb:17:in `run'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/runner.rb:16:in `block in run_one'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/errors.rb:5:in `handle_exception'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/runner.rb:15:in `run_one'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/runner.rb:31:in `block in run_all'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/runner.rb:30:in `each'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/runner.rb:30:in `run_all'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/active_record_doctor-1.11.0/lib/active_record_doctor/rake/task.rb:60:in `block in define'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `block in execute'
/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `each'
```

**Comment by fatkodima** [comment]
Try to debug with the debugger. `lhs.columns` should return an array, and `.count` is called on the array. Maybe this method is redefined somewhere else. Try to see  what `lhs.columns.method(:count).source_location` returns. Should be `nil`, if not overridden.

**Comment by owst** [comment]
We have hit this problem too for a table with a multi-column (expression) index in Postgres. 

The following shows that such an index returns a String from `columns`:
```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  gem "activerecord", "~> 7.0.0"
  gem "pg"
end

require "active_record"

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(
  adapter: "postgresql",
  database: "ard_bug",
  username: ENV.fetch("PGUSER"),
  password: ENV.fetch("PGPASSWORD")
)
ActiveRecord::Schema.define do
  create_table :people, force: true do |t|
    t.string :first_name
    t.string :last_name

    t.index :first_name
    t.index :last_name
    t.index "lower(last_name), lower(first_name)"
  end
end

class Person < ActiveRecord::Base
end

indexes = Person.connection.indexes("people")
indexes.map(&:columns).each do |index|
  puts "index: #{index.inspect}, class: #{index.class}"
end
```

which prints:
```
index: ["first_name"], class: Array
index: ["last_name"], class: Array
index: "lower((last_name)::text), lower((first_name)::text)", class: String
```

**Comment by fatkodima** [comment]
Should be fixed by https://github.com/gregnavis/active_record_doctor/pull/135

**Comment by gregnavis** [comment]
Fixed in 1.12.0.
